File: presentation/src/main/java/whenyourcar/presentation/controller/user/UserCommonController.java
package whenyourcar.presentation.controller.user;

import com.fasterxml.jackson.core.JsonProcessingException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpSession;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import whenyourcar.domain.common.code.status.SuccessStatus;
import whenyourcar.domain.user.dto.security.SessionUser;
import whenyourcar.presentation.apiPayload.ApiResponse;
import whenyourcar.presentation.facade.user.UserCommonFacade;

import java.util.Enumeration;
import java.util.HashMap;
import java.util.Map;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/user")
public class UserCommonController {
    private final UserCommonFacade userCommonFacade;

    @GetMapping("/profile")
    public ApiResponse<SessionUser> getProfile(HttpServletRequest request)  {
        return ApiResponse.onSuccess(SuccessStatus.USER_PROFILE_SUCCESS, userCommonFacade.getUserProfile(request));
    }
}

File: api/src/main/kotlin/howmuchcar/api/controller/user/UserCommonController.kt
package howmuchcar.api.controller.user

import howmuchcar.api.security.annotation.CurrentUser
import howmuchcar.application.dto.user.UserProfileResponse
import howmuchcar.application.facade.user.UserCommonFacade
import howmuchcar.common.apiPayload.ApiResponse
import howmuchcar.domain.entity.User
import lombok.RequiredArgsConstructor
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/user")
class UserCommonController (
    private val userCommonFacade: UserCommonFacade
){
    @GetMapping("/profile")
    fun getUserProfile(@CurrentUser user: User) : ApiResponse<UserProfileResponse> {
        return ApiResponse.onSuccess(userCommonFacade.getUserProfile(user))
    }
}